src/funcs.c

```diff
@@ -57,6 +57,36 @@
 
 #include <src/hookintrprtr.h>
 
+static ModuleStateOffset funcsStateOffset;
+typedef struct {
+    Int RecursionDepth;
+    Obj ExecState;
+} funcsModuleState;
+
+void IncRecursionDepth(void)
+{
+    MODULE_STATE(funcs, RecursionDepth)++;
+}
+
+void DecRecursionDepth(void)
+{
+    MODULE_STATE(funcs, RecursionDepth)--;
+    /* Apparently this is wrong
+    GAP_ASSERT(MODULE_STATE(RecursionDepth) >= 0);
```

Comment:
I am not sure it is "wrong", but maybe it is currently violated -- which smells like a bug to me.